src/main/c/NativePtr.cpp

```diff
@@ -0,0 +1,366 @@
+#include "hail/NativePtr.h"
+#include "hail/NativeObj.h"
+#include <jni.h>
+#include <assert.h>
+#include <stdio.h>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace hail {
+
+// JNI interface functions corresponding to Scala NativeBase and NativePtr
+
+#define NATIVEFUNC(typ) \
+  extern "C" JNIEXPORT typ JNICALL
+
+namespace {
+
+class AlignedBuf {
+public:
+  char buf_[2*sizeof(long)];
+  long force_align_;
```

Comment:
To force the object to be 8-byte aligned

Reply:
I think this can be accomplished more explicitly with
```C++
alignas(8) char buf_[2*sizeof(long)];
```
```C++
alignas(long) char buf_[2*sizeof(long)];
```
or
```C++
std::aligned_storage_t<2*sizeof(long), alignof(long)> buf;
```
or on the class
```
class alignas(long) AlignedBuf {
```
See http://en.cppreference.com/w/cpp/language/alignas, https://en.cppreference.com/w/cpp/types/aligned_storage. I think any of these are fine, and I would prefer to make the alignment restriction explicit using standard language features.

Reply:
Never knew that existed, but ok.